app/models.py

```diff
@@ -0,0 +1,230 @@
+from peewee import *
+
+DATABASE_NAME = "database.db"
+database = SqliteDatabase(DATABASE_NAME)
+
+
+# Database models
+class BaseModel(Model):
+    class Meta:
+        database = database
+
+
+class User(BaseModel):
+    username = CharField(unique=True)
+    fullname = CharField()
+    mail_address = CharField()
+    password = CharField()
+    user_type = CharField()
+
+    def is_student(self):
+        return User.user_type == "is_student"
+
+    def is_teacher(self):
+        return User.user_type == "teacher"
+
+    def get_user_courses(self):
+        return (Course
+                .select()
+                .join(UserCourseRelationship, on=UserCourseRelationship.course)
+                .where(UserCourseRelationship.user == self)
+                .order_by(Course.id)
+                )
+
+
+class Course(BaseModel):
+    course_id = IntegerField(unique=True)
+    name = CharField()
+    open_date = DateTimeField()
+
+    def get_course_administrators(self):
+        return (User
+                .select()
+                .join(UserCourseRelationship, on=UserCourseRelationship.user)
+                .where((UserCourseRelationship.course == self) & UserCourseRelationship.is_admin))
+
+
+class UserCourseRelationship(BaseModel):
+    user = ForeignKeyField(User, backref='user')
+    course = ForeignKeyField(Course, backref='course')
+    is_admin = BooleanField(default=False)
+
+
+class Lecture(BaseModel):
+    lecture_id = IntegerField(unique=True)
+    course = ForeignKeyField(Course, backref='lectures')
+    date = DateTimeField()
+    subject = CharField()
+
+    def get_participants(self):
+        return (User
+                .select()
+                .join(UserLectureRelationship, on=UserLectureRelationship.user)
+                .where(UserLectureRelationship.lecture == self))
+
+
+class UserLectureRelationship(BaseModel):
+    user = ForeignKeyField(User, backref='user')
+    lecture = ForeignKeyField(Lecture, backref='lecture')
+
+
+# Utility to create the tables
+def create_tables():
+    with database:
+        database.create_tables([User, Course, Lecture, UserCourseRelationship, UserLectureRelationship])
+
+
+# Examples
+def get_object(model, *expressions):
+    try:
+        return model.get(*expressions)
+    except:
+        raise IntegrityError()
```

Comment:
Why raise `IntegrityError` instead of original one?
This also means we are missing the original exception with it's message

Reply:
True that, will change :)